mysql-test/suite/innodb/include/temp_table_encrypt.inc

```diff
@@ -0,0 +1,158 @@
+# test temporary tables encryption
+
+--let $MYSQL_DATA_DIR= `select @@datadir`
+
+# we need restart to make sure keyring settings are picked up
+let $restart_hide_args = 1;
+let $restart_parameters = $KEYRING_RESTART_PARAM --innodb_max_dirty_pages_pct=0;
+--source include/restart_mysqld.inc
+
+CREATE TEMPORARY TABLE t01 (a TEXT) ENGINE=InnoDB;
+INSERT INTO t01 VALUES ('Lorem ipsum dolor sit amet, consectetur adipiscing elit');
+
+# wait until pages flushed
+--let $wait_condition= SELECT COUNT(*) = 0 FROM performance_schema.global_status WHERE LOWER(variable_name) = 'innodb_buffer_pool_pages_dirty' AND variable_value > 0
```

Comment:
Looks like a typo: isn't wait_condition supposed to be wait *termination* condition, i.e. value = 0 ?

Reply:
Actual condition here is `COUNT(*) = 0`, i.e. there is no variable with name `innodb_buffer_pool_pages_dirty` and `value > 0`

Reply:
I'll rewrite it to be more radable